remix-astwalker/src/types.ts

```diff
@@ -0,0 +1,22 @@
+export interface Node {
+  ast: any | null,
```

Comment:
Please please please, add a detailed type for `ast` ⚡️! This would really leverage the power of Typescript.

Reply:
this is using the `legacyAST` which will be at some point completely dropped by the solidity compiler. So it's be better to `Type the new `AST` directly. 
Is there some reference, documentation or code we can use for "typing" all the nodes available in the new AST? @chriseth @axic 
for reference, here is the code for the support of new AST: https://github.com/ConsenSys/truffle-security/blob/dcd862baab6b94bd7a8fe2d9caeee69dc235009b/compat/remix-lib/astWalker.js#L2
we should import it in our codebase

Reply:
I took a look. Only major changes were in `manageCallback()`
```
if (node.nodeType in callback) {
    return callback[node.nodeType](node);
}
```
vs
```
if (node.name in callback) {
  return callback[node.name](node)
}
```
We are taking a simple example and trying to build the AST type interface. Need to look at documentations of new AST specification.

Reply:
> Is there some reference, documentation or code we can use for "typing" all the nodes available in the new AST? @chriseth @axic

These information could help us a lot.